bridge/packages/feature/home/lib/src/home_view.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:math' as math;
+
+import 'package:core_bloc/bloc.dart';
+import 'package:core_data/data.dart';
+import 'package:core_designsystem/designsystem.dart';
+import 'package:core_ui/ui.dart';
+import 'package:feature_home/src/bloc/home_bloc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class HomeView extends StatelessWidget {
+  const HomeView({
+    super.key,
+    required this.onNavAuthRequest,
+    required this.onNavJoinGameRequest,
+    required this.onNavHostGameRequest,
+  });
+
+  final VoidCallback? onNavAuthRequest;
+  final VoidCallback? onNavJoinGameRequest;
+  final void Function(String?)? onNavHostGameRequest;
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiBlocListener(
+      listeners: [
+        BlocListener<HomeBloc, HomeState>(
+          listener: (context, state) {
+            // TODO(home): handle errors.
+            final roomId = state.roomId;
+            if (roomId.isNotEmpty) {
+              context.read<HomeBloc>().add(HomeRoomIdCleaned());
+              onNavHostGameRequest?.call(roomId);
+            }
+          },
+        ),
+        BlocListener<AuthenticationBloc, AuthenticationState>(
+          listener: (context, state) {
+            if (state.status == AuthenticationStatus.unauthenticated) {
+              onNavAuthRequest?.call();
+            }
+          },
+        ),
+      ],
+      child: Container(
+        alignment: Alignment.center,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Stack(
```

Comment:
You could make this more readable by extracting widgets. For example, this Stack can be extracted into MainGraphic widget.